File: src/test/java/net/sf/rails/game/state/OwnableItemImpl.java
package net.sf.rails.game.state;

import net.sf.rails.game.state.Item;
import net.sf.rails.game.state.Ownable;
import net.sf.rails.game.state.OwnableItem;

/**
 * Test implementation of OwnableItem
 */
class OwnableItemImpl extends OwnableItem<Ownable> {

    protected OwnableItemImpl(Item parent, String id) {
        super(parent, id, Ownable.class);
    }

    static OwnableItemImpl create(Item parent, String id) {
        return new OwnableItemImpl(parent, id);
    }
    
}

File: dotnet/GameStateTest/OwnableItemImpl.cs
using GameLib.Net.Game.State;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

/**
 * Test implementation of OwnableItem

 */
namespace GameStateTest
{
    public class OwnableItemImpl : OwnableItem<IOwnable>
    {
        protected OwnableItemImpl(IItem parent, string id) : base(parent, id)
        {

        }

        public static OwnableItemImpl Create(IItem parent, string id)
        {
            return new OwnableItemImpl(parent, id);
        }
    }
}
